# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: third_party/rust/rental/tests/clone.rs
# Commit: 16350e08d34f
# Full Hash: 16350e08d34ff06c7f46745c6b160c7a7bda9140
# Author: Dana Keeler <dkeeler@mozilla.com>
# Date: 2019-12-06 04:43:57
# Regressor Bug: 1586855
# File Overlap Count: 6
# Description:
#   bug 1586855 - incorporate CRLite filters into cert_storage r=jcj,kjacobs
#   
#   This patch implements CRLite lookups for TLS server certificate revocation
#   information in telemetry-only mode. It adds a new preference
#   "security.pki.crlite_mode" to control the behavior of this feature. Setting
# ==============================================================================

diff -r 64fd40663930 -r 16350e08d34f third_party/rust/rental/tests/clone.rs
--- /dev/null	Thu Jan 01 00:00:00 1970 +0000
+++ b/third_party/rust/rental/tests/clone.rs	Thu Dec 05 22:41:53 2019 +0000
@@ -0,0 +1,50 @@
+#[macro_use]
+extern crate rental;
+
+
+pub struct Foo {
+	i: i32,
+}
+
+pub struct Bar<'i> {
+	iref: &'i i32,
+	misc: i32,
+}
+
+impl <'i> Clone for Bar<'i> {
+	fn clone (&self) -> Self {
+		Bar{
+			iref: Clone::clone(&self.iref),
+			misc: Clone::clone(&self.misc),
+		}
+	}
+}
+
+
+rental! {
+	mod rentals {
+		use super::*;
+		use std::sync::Arc;
+
+		#[rental(clone)]
+		pub struct FooClone {
+			foo: Arc<Foo>,
+			fr: Bar<'foo>,
+		}
+	}
+}
+
+
+#[test]
+fn clone() {
+	use std::sync::Arc;
+
+	let foo = Foo { i: 5 };
+	let rf = rentals::FooClone::new(Arc::new(foo), |foo| Bar{ iref: &foo.i, misc: 12 });
+	assert_eq!(5, rf.rent(|f| *f.iref));
+
+	let rfc = rf.clone();
+	assert_eq!(5, rfc.rent(|f| *f.iref));
+}
+
+